rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java

```diff
@@ -143,6 +143,24 @@ public void logCommitFederation(Block executionBlock, Federation oldFederation,
         addLog(encodedTopics, encodedData);
     }
 
+    @Override
+    public void logCommitFederationFailure(Block executionBlock, Federation proposedFederation) {
+        CallTransaction.Function event = BridgeEvents.COMMIT_FEDERATION_FAILED.getEvent();
+
+        byte[][] encodedTopicsSerialized = event.encodeEventTopics();
+        List<DataWord> encodedTopics = getEncodedTopics(encodedTopicsSerialized);
+
+        byte[] proposedFederationRedeemScriptSerialized = proposedFederation.getRedeemScript().getProgram();
+        long executionBlockNumber = executionBlock.getNumber();
+
+        byte[] encodedData = event.encodeEventData(
+            proposedFederationRedeemScriptSerialized,
```

Comment:
```suggestion
        Script proposedFederationRedeemScript = proposedFederation.getRedeemScript();
        long executionBlockNumber = executionBlock.getNumber();

        byte[] encodedData = event.encodeEventData(
            proposedFederationRedeemScript.getProgram(),
```
Does this way respect Demeters law or is it just cheating?

Reply:
jajaja i think is just cheating. To respect it I think we should have a `proposedFederation.getRedeemScriptProgram()`